nixos/modules/services/security/oauth2_proxy_nginx.nix

```diff
@@ -21,22 +34,26 @@ in
       '';
     };
   };
+
   config.services.oauth2_proxy = mkIf (cfg.virtualHosts != [] && (hasPrefix "127.0.0.1:" cfg.proxy)) {
     enable = true;
   };
-  config.services.nginx = mkIf config.services.oauth2_proxy.enable (mkMerge
-  ((optional (cfg.virtualHosts != []) {
-    recommendedProxySettings = true; # needed because duplicate headers
-  }) ++ (map (vhost: {
-    virtualHosts.${vhost} = {
-      locations."/oauth2/" = {
+
+  config.services.nginx = mkIf config.services.oauth2_proxy.enable (mkMerge ([
+    {
+      virtualHosts.${cfg.domain}.locations."/oauth2/" = {
```

Comment:
This will cause an eval error (via the `lib.head` default) if both nginx and oauth2_proxy are enabled, and the oauth2_proxy.nginx backend is unused. What if the user wants to configure oauth2_proxy in caddy while reverse proxying to nginx due to some php service with a bunch of route rules.

Reply:
I've removed the default